src/workflows/registration.ts

```diff
@@ -113,3 +116,18 @@ export async function getSignableRegistrationOnchain(
     payload_hash: response.data.payload_hash,
   };
 }
+
+async function isUserRegistered(
+  userAddress: string,
+  usersApi: UsersApi,
+): Promise<boolean> {
+  try {
+    await usersApi.getUsers({ user: userAddress });
+  } catch (error) {
+    const axiosError = error as AxiosError;
+    if (axiosError.response?.status === 404) {
+      return false;
+    }
+  }
+  return true;
```

Comment:
If getUsers throws errors and it is not 404 you return true from isUserRegistered and that ends the function with no error? It feels wrong.

Reply:
@zatlite The [first implementation](https://github.com/immutable/imx-core-sdk/pull/85/commits/f34e2d44ff628b3f53d8de6a52d1b4b4c44e003c) was throwing the error in case it wasn't a 404, which is expected.
But we decided to change it because this check shouldn't make the workflow fail, since it's just a way to finish the process earlier.
Do you think that we should throw the exception and handle it inside the workflow?